services/m3ctl/main/main.go

```diff
@@ -100,6 +108,20 @@ func main() {
 		os.Exit(1)
 	}
 
+	envPort := os.Getenv(portEnvVar)
+
+	if envPort != "" {
+		if p, err := strconv.Atoi(envPort); err == nil {
+			cfg.HTTP.Port = p
```

Comment:
nit: log a message indicating you're overriding the config value with an env var